### PR TITLE
Fixes #7273: if compilation flags tests fails, rpm build fails

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/7273

### CODE REVIEW

File: rudder-agent/SPECS/rudder-agent.spec [hunk 1]
```diff
@@ -301,8 +301,7 @@ TRY_CFLAGS="-fPIE -fstack-protector"
 
 FLAG_TEST_FILE=`mktemp /tmp/hardening.XXXXXX`
 echo "void main() {}" > "${FLAG_TEST_FILE}.c"
-gcc ${TRY_CFLAGS} ${TRY_LDFLAGS} -o "${FLAG_TEST_FILE}" "${FLAG_TEST_FILE}.c" 2>/dev/null
-if [ $? -eq 0 ]
+if [ $(gcc ${TRY_CFLAGS} ${TRY_LDFLAGS} -o "${FLAG_TEST_FILE}" "${FLAG_TEST_FILE}.c" >/dev/null 2>&1; echo $?) -eq 0 ]
```
Comment (peckpeck) [inline rudder-agent/SPECS/rudder-agent.spec]: I think the problem is set -e and this scheme will fail with set -e too.
Instead use: if gcc ..... ; then true; else ... ; fi

**Comment by VinceMacBuche** [comment]
PR updated

File: rudder-agent/SPECS/rudder-agent.spec (hunk 1 above)
Comment (VinceMacBuche) [inline rudder-agent/SPECS/rudder-agent.spec]: It does not fail in set -e 

I tried with the following script: 

     #!/bin/sh

      set -e

    SECURE_CFLAGS="NO FLAGS"
    TRY_LDFLAGS="-pie -Wl,-z,relro -Wl,-z,now"
    TRY_CFLAGS="-fPIE -fstack-protector"
    FLAG_TEST_FILE=`mktemp /tmp/hardening.XXXXXX`
    echo "void main() {}" > "${FLAG_TEST_FILE}.c"
    if gcc ${TRY_CFLAGS} ${TRY_LDFLAGS} -o "${FLAG_TEST_FILE}" "${FLAG_TEST_FILE}.c" >/dev/null 2>&1
     then
       SECURE_CFLAGS="${TRY_CFLAGS}"
       SECURE_LDFLAGS="${TRY_LDFLAGS}"
    fi
    echo $SECURE_CFLAGS

**Comment by VinceMacBuche** [comment]
PR updated

**Comment by peckpeck** [comment]
OK Good

**Pull request merged**